core/http/src/main/java/org/trellisldp/http/impl/GetHandler.java

```diff
@@ -355,29 +356,22 @@ private ResponseBuilder getLdpNr(final ResponseBuilder builder) {
 
         // Short circuit HEAD requests
         if (HEAD.equals(getRequest().getMethod())) {
-            return builder;
+            return completedFuture(builder);
         }
 
         // Stream the binary content
-        final StreamingOutput stream = new StreamingOutput() {
-            @Override
-            public void write(final OutputStream out) throws IOException {
-                // TODO -- with JDK 9 use InputStream::transferTo instead of IOUtils::copy
-                try (final InputStream binary = getBinaryStream(dsid, getRequest()).toCompletableFuture().join()) {
-                    IOUtils.copy(binary, out);
-                }
-            }
-        };
-
-        return builder.entity(stream);
+        // TODO -- with JDK 9 use InputStream::transferTo instead of IOUtils::copy
+        return getBinaryStream(dsid, getRequest())
+                        .thenApply(in -> (StreamingOutput) out -> IOUtils.copy(in, out))
```

Comment:
Should the input stream be closed here?

Reply:
Hm, I think you might be right. I'll take a look.

Reply:
Good catch. I've fixed it.